### PR TITLE
Grid task log, multi select and ts files migration.

### PR DESCRIPTION
**Pull request by pierrejeambrun**
Followup on https://github.com/apache/airflow/pull/24403.

- Adding support for multi select on the log level (via [chakra-react-select](https://www.npmjs.com/package/chakra-react-select))
- Migrating some files to ts

@bbovenzi , we discussed about removing colors in the drop down. Adding colors inside the logs is not trivial. Even without colors in the logs, I think the left border in the drop down makes it easier to identify the level we are looking for.

I will happily change the style depending on your feedback.

Edit: Converting to draft, not ready. I would like to also migrate associated test file to ts in this PR.

image

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragement file, named `{pr_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/www/static/js/grid/api/useTaskLog.tsx [hunk 1]
```diff
@@ -17,23 +17,31 @@
...
+  fullContent: boolean,
 }) => {
-  const url = taskLogApi.replace('_DAG_RUN_ID_', dagRunId).replace('_TASK_ID_', taskId).replace(/-1$/, taskTryNumber);
+  let url: string = '';
+  if (taskLogApi) {
+    url = taskLogApi.replace('_DAG_RUN_ID_', dagRunId).replace('_TASK_ID_', taskId).replace(/-1$/, taskTryNumber.toString());
+  }
 
-  return useQuery(
+  return useQuery<string, Error>(
```
Comment (bbovenzi) [inline airflow/www/static/js/grid/api/useTaskLog.tsx]: Let's avoid setting any types directly on `useQuery` I think I accidentally have it on `useTasks()` which I'll fix

Comment (pierrejeambrun) [reply airflow/www/static/js/grid/api/useTaskLog.tsx]: Done

**Comment by bbovenzi** [comment]
Screen Shot 2022-06-27 at 5 14 32 PM

The font colors aren't matching.

Don't we want the multi-select for the file source too?

**Comment by bbovenzi** [comment]
Let's update the `scrollIntoView()` effect to be based off of the `parsedLogs` and not data. The scroll position is lost when filtering.

**Review by bbovenzi — APPROVED** [review]
nice work! this keeps becoming a more useful view

**Pull request merged**